src/ray/common/ray_syncer/ray_syncer.cc

```diff
@@ -231,19 +230,21 @@ void RaySyncer::Connect(const std::string &node_id,
                                     << NodeID::FromBinary(node_id);
                       Connect(node_id, channel);
                     },
-                    /* delay_duration = */ std::chrono::milliseconds(2000));
+                    /* delay_microseconds = */ std::chrono::milliseconds(2000));
+              } else {
+                node_state_->RemoveNode(node_id);
               }
             },
             /* stub */ std::move(stub));
         Connect(reactor);
         reactor->StartCall();
-      },
-      "");
+      }))
+      .get();
```

Comment:
Does it mean this will be a blocking call?

Reply:
Yes, it'll. It returns a future.

Reply:
Will it affect perf like blocking the thread for too long?